kbs/docs/id-key.md

```diff
@@ -0,0 +1,84 @@
+# ID-key Plugin
+
+In addition to fetchable resources upon successful attestation, KBS also offers the ID-key plugin for encrypting and decrypting data with a key provisioned by the KBS. This key can be persisted across multiple reboots of CVMs and help tie decryption resources to a specific KBS instance.
+
+## Data Encryption
+
+As a guest owner would be responsible for encrypting persistent storage, they are the sole authority that is permitted to request resources be encrypted. The KBS is expected to be fully in control by the guest owner with credentials configured.
+
+## Flow
+
+### Provisioning encrypted data
+
+A credentialed guest owner is the only one authorized to request that keys be encrypted by the ID-key plugin. To do this, the guest owner will POST some plaintext data (created locally) to be encrypted. This plaintext data could be a LUKS passphrase, SecureBoot keys, etc. The actual purpose of the data is of no concern to the KBS.
```

Comment:
can you specify guest-owner? would the POST endpoint be called from within a CVM? or would a this be performed by someone who has admin access to Trustee?

Reply:
I use "guest owner" as the one who has admin access to Trustee, and will be running their "owned" CVMs on an untrusted platform.